vncviewer/win32TouchHandler.cxx

```diff
@@ -0,0 +1,251 @@
+/* Copyright 2020 Samuel Mannehed for Cendio AB
+ *
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "i18n.h"
+#include "win32TouchHandler.h"
+
+#define XK_MISCELLANY
+#include <rfb/keysymdef.h>
+#include <rfb/LogWriter.h>
+static rfb::LogWriter vlog("win32TouchHandler");
+
+const DWORD MOUSEMOVE_FLAGS = MOUSEEVENTF_MOVE | MOUSEEVENTF_ABSOLUTE |
+                              MOUSEEVENTF_VIRTUALDESK;
+
+static void dbgPrint(DWORD id, GHEvent ghev) {
+  const char* action;
+
+  switch (id) {
+
+  case GID_BEGIN:        action = "Gesture BEGIN";        break;
+  case GID_ZOOM:         action = "Gesture ZOOM";         break;
+  case GID_PAN:          action = "Gesture PAN";          break;
+  case GID_TWOFINGERTAP: action = "Gesture TWOFINGERTAP"; break;
+  case GID_END:          action = "Gesture END";          break;
+
+  case WM_MOUSEMOVE:     action = "fake MOUSEMOVE";       break;
+  case WM_LBUTTONDOWN:   action = "fake LBUTTONDOWN";     break;
+  case WM_LBUTTONUP:     action = "fake LBUTTONUP";       break;
+  case WM_RBUTTONDOWN:   action = "fake RBUTTONDOWN";     break;
+  case WM_RBUTTONUP:     action = "fake RBUTTONUP";       break;
+  case WM_MBUTTONDOWN:   action = "fake MBUTTONDOWN";     break;
+  case WM_MBUTTONUP:     action = "fake MBUTTONUP";       break;
+  case WM_MOUSEWHEEL:    action = "fake MOUSEWHEEL";      break;
+  case WM_MOUSEHWHEEL:   action = "fake MOUSEHWHEEL";     break;
+  case WM_KEYDOWN:       action = "fake KEYDOWN";         break;
+  case WM_KEYUP:         action = "fake KEYUP";           break;
+
+  default:               action = "Unknown event";        break;
+  }
+
+  vlog.debug("%s (%i, %i)", action, (int)ghev.event_x, (int)ghev.event_y);
+}
+
+win32TouchHandler::win32TouchHandler(HWND hWnd) :
```

Comment:
Should be `Win32TouchHandler` (and rename files to match).

Reply:
Fixed in b30a25ef030a56cb5e7453e430525a257a775090